apps/easypid/src/features/scan/FunkeQrScannerScreen.tsx

```diff
@@ -0,0 +1,283 @@
+        {showMyQrCode && <FunkeQrOverlay qrCodeData={qrCodeData} />}
+      </AnimatedStack>
+
+      {isLoading && (
+        <Page
+          position="absolute"
+          jc="center"
+          ai="center"
+          g="md"
+          enterStyle={{ opacity: 0, y: 50 }}
+          exitStyle={{ opacity: 0, y: -20 }}
+          y={0}
+          bg="$black"
+          opacity={0.7}
+          animation="lazy"
+        >
+          <Spinner variant="dark" />
+          <Paragraph variant="sub" color="$white" textAlign="center">
+            Loading invitation
+          </Paragraph>
+        </Page>
+      )}
+      <Page
+        bg="transparent"
+        position="absolute"
+        alignItems="center"
+        justifyContent="flex-end"
+        bottom={0}
+        left={0}
+        right={0}
+        pb={bottom * 2}
+      >
+        <AnimatedStack
+          alignItems="center"
+          layout={useSpringify(LinearTransition)}
+          onPress={handleQrButtonPress}
+          bg="$grey-100"
+          br="$12"
+          py="$2.5"
+          px="$5"
+          accessibilityRole="button"
+          aria-label={showMyQrCode ? 'Scan QR code' : 'Show my QR code'}
+          minWidth={150}
+        >
+          <AnimatedStack
+            key={showMyQrCode ? 'scan-qr-code' : 'show-my-qr-code'}
+            entering={FadeIn.delay(200).duration(200)}
+            exiting={FadeOut.duration(200)}
+            flexDirection="row"
+            ai="center"
+            gap="$2"
+          >
+            <Paragraph fontWeight="$semiBold" ai="center" ta="center" color="$grey-900">
+              {showMyQrCode ? 'Scan QR code' : 'Show my QR code'}
+            </Paragraph>
+            {showMyQrCode ? (
+              <HeroIcons.CameraFilled size={20} color="$grey-900" />
+            ) : (
+              <HeroIcons.QrCode size={20} color="$grey-900" />
+            )}
+          </AnimatedStack>
+        </AnimatedStack>
+      </Page>
+    </>
+  )
+}
+
+function FunkeQrOverlay({ qrCodeData }: { qrCodeData?: string }) {
+  const { width } = useWindowDimensions()
+  const { bottom, top } = useSafeAreaInsets()
+  const { withHaptics } = useHaptics()
+  const { replace } = useRouter()
+
+  useEffect(() => {
+    if (qrCodeData) {
+      void waitForDeviceRequest().then((data) => {
+        if (data) {
+          // Take the Doc item that matches with the mdoc pid type
+          // Only support one doc item for now
+
+          const requestedNamespace = data.requestedItems[0][pidSchemes.msoMdocDoctypes[0]]
+          if (!requestedNamespace) throw new Error('Unsupported credential requested.')
+
+          pushToOfflinePresentation({
+            sessionTranscript: Buffer.from(data.sessionTranscript).toString('base64'),
+            deviceRequest: Buffer.from(data.deviceRequest).toString('base64'),
+            requestedAttributes: Object.entries(requestedNamespace).map(([key]) => key),
```

Comment:
do we only handle the requested namespaces (not attributes within?)

Reply:
It takes the requestedNamespace which is equal to the pid, and in there are the attributes requested.